docs: instructions to solve v10 compatibility issues with jest

pkuczynski opened the pull request.

pkuczynski committed:
docs: instructions to solve v10 compatibility issues with jest

xDivisionByZerox commented on docs/guide/upgrading.md:
````diff
@@ -47,11 +47,37 @@ As mentioned in the previous section, CJS can still be used if you use a modern
+
+When using Jest in a TypeScript project you might be already using `ts-jest` for transforming `ts` files on the fly. In such case you should apply following changes to your `jest.config.ts`:
+
+```ts
+// Transform both `ts` and `js` files. Defining only `ts` would not be enough, as we also need to transform @faker-js
+transform: {
+  '^.+\\.(t|j)s$': 'ts-jest',
+
+  // or when you pass more settings:
+  '^.+\\.(t|j)s$': [
+    'ts-jest',
+    // ... other setttings
+  ]
+}
+
+// Exclude from transformation all files in `node_modules`, except `@faker-js`
+transformIgnorePatterns: [
+  // npm
+  'node_modules/(?!@faker-js).+',
+
+  // pnpm
+  'node_modules/.pnpm/.+/node_modules/(?!@faker-js).+'
+],
+```
````
The change itself looks fine. I was thinking about representing the code block for the config as a [Code Group](https://vitepress.dev/guide/markdown#code-groups) to better differentiate between npm and pnpm packages. 
BUT, this documentation only represents a migration option for `ts-jest`. By default Jest uses [babel-jest](https://github.com/jestjs/jest/tree/main/packages/babel-jest#setup). Also, the transformer configuration itself is not that important - only that you have one defined for `*.js` files. That's why I would probably replace the "transform" section in a more generic way, like this:

```ts
// Transform both `ts` and `js` files. Defining only `ts` would not be enough, as we also need to transform @faker-js
transform: {
  '^.+\\.(t|j)s$': 'ts-jest', // use the transformer and configuration of your choice
}

// Exclude from transformation all files in `node_modules`, except `@faker-js`
transformIgnorePatterns: [
  // npm
  'node_modules/(?!@faker-js).+',

  // pnpm
  'node_modules/.pnpm/.+/node_modules/(?!@faker-js).+'
],
```

ST-DDT replied:
I think the code groups would be worth exploring for this case.
Although I would consider them optional, because we haven't used them much elsewhere AFAIK.

pkuczynski replied:
@xDivisionByZerox not sure about it being more generic makes a lot of sense. I think most of the people would be able to easily follow the diff between npm, pnpm, etc. And I do not know how to use code groups? If you show me example, I can certainly do that, although not sure if this is worth the hassle...

xDivisionByZerox replied:
All good. Since this has been open for a while I'm fine with the implementation as is. First get the helping documentation out so people have an easier time. Additional UX features can be added in a later iteration. :)

pkuczynski replied:
Yeah, I have should been more active on this. Apologies mate... Too much on my plate recently...

xDivisionByZerox replied:
All good. :v:
This more comment on my constant nitpicking towards myself. I'm sorry if you interpreted it as comment towards you.

matthewmayer commented:
Maybe we should try to keep this short and put a longer description and workarounds on https://fakerjs.dev/guide/frameworks.html

xDivisionByZerox commented:
@pkuczynski are you still interested in providing a solution here?

ST-DDT commented:
What is the status on this PR?

xDivisionByZerox commented:
I will close this as stale.

pkuczynski commented:
Hey @xDivisionByZerox and @ST-DDT - sorry for late reply. Not sure what you wanted me to do here? This documentation change felt complete from my POV?

ST-DDT commented on docs/guide/upgrading.md:
```diff
@@ -47,11 +47,37 @@ As mentioned in the previous section, CJS can still be used if you use a modern
+  // or when you pass more settings:
+  '^.+\\.(t|j)s$': [
+    'ts-jest',
+    // ... other setttings
```
Our playground seems to specify slightly more parameters depending on the setup:

https://github.com/faker-js/playground/blob/main/playgrounds/jest-cjs/jest.config.js

If these are required for using faker, then we should mention them here as well.

Maybe also add a link to the playground itself.

pkuczynski replied:
The setup I posted was a minimal configuration which worked with my project. Not sure, mayne the other settings are more tailored to a different setup or added later, after I created my original PR?

Nonetheless, I linked the playground now...

pkuczynski committed:
Update upgrading.md with playground link and upgrade advice

Added a link to the playground for reference and provided guidance on upgrading from v9 to v10.

xDivisionByZerox commented:
> Maybe we should try to keep this short and put a longer description and workarounds on [fakerjs.dev/guide/frameworks.html](https://fakerjs.dev/guide/frameworks.html)

I think this is worth exploring, since new users will most likely check this section instead of the migration guide. I'll provide a follow up PR that provides additional context in the frameworks section.
